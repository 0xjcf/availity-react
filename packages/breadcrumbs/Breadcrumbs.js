import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumb, BreadcrumbItem } from 'reactstrap';

const Breadcrumbs = ({ crumbs, active, emptyState, ...props }) => (
  <Breadcrumb {...props}>
    <BreadcrumbItem>
      <a href="/public/apps/dashboard">Home</a>
    </BreadcrumbItem>

    {crumbs &&
      crumbs.length > 0 &&
      crumbs.map(crumb => (
        <BreadcrumbItem key={crumb.name + crumb.url}>
          {crumb.name && crumb.url ? (
            <a href={crumb.url}>{crumb.name}</a>
          ) : (
            <span>{emptyState}</span>
          )}
        </BreadcrumbItem>
      ))}

    <BreadcrumbItem active>{active || emptyState}</BreadcrumbItem>
  </Breadcrumb>
);

Breadcrumbs.propTypes = {
  crumbs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      url: PropTypes.string,
    })
  ),
  active: PropTypes.string.isRequired,
  emptyState: PropTypes.string,
};

Breadcrumbs.defaultProps = {
  emptyState: 'â€¦',
};

export default Breadcrumbs;
