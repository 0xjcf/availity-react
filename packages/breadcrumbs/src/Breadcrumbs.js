import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumb, BreadcrumbItem } from 'reactstrap';

const Breadcrumbs = ({ crumbs, active, emptyState, children, linkTag: LinkTag, homeUrl, ...rest }) => {
  const renderBreadCrumb = (crumb) => {
    // default breadcrumbitem render
    let breadCrumbItemChildren = <span>{emptyState}</span>;
    // render static links
    if (crumb.name && crumb.url) {
      breadCrumbItemChildren = (
        <LinkTag aria-label={crumb.name} href={crumb.url}>
          {crumb.name}
        </LinkTag>
      );
    }
    return <BreadcrumbItem key={crumb.name + crumb.url}>{breadCrumbItemChildren}</BreadcrumbItem>;
  };

  return (
    <Breadcrumb {...rest}>
      <BreadcrumbItem>
        <LinkTag aria-label="Home" href={homeUrl}>
          Home
        </LinkTag>
      </BreadcrumbItem>
      {crumbs && crumbs.length > 0 && crumbs.map((crumb) => renderBreadCrumb(crumb))}
      {children}
      <BreadcrumbItem active>{active || emptyState}</BreadcrumbItem>
    </Breadcrumb>
  );
};

Breadcrumbs.propTypes = {
  /** The currently active page */
  active: PropTypes.string.isRequired,
  /** Additional parent breadcrumbs */
  crumbs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      url: PropTypes.string,
    })
  ),
  /** Tag for link */
  linkTag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  /** Breadcrumb value when state is empty */
  emptyState: PropTypes.string,
  /** Node to go between parent and active breadcrumb */
  children: PropTypes.node,
  /** URL for home breadcrumb */
  homeUrl: PropTypes.string,
};

Breadcrumbs.defaultProps = {
  emptyState: 'â€¦',
  homeUrl: '/public/apps/dashboard',
  linkTag: 'a',
};

export default Breadcrumbs;
