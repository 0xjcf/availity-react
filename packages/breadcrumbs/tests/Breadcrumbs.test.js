import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumbs from '..';

describe('Breadcrumbs', () => {
  test('should render', () => {
    const component = renderer.create(<Breadcrumbs active="" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render child as current page', () => {
    const component = renderer.create(<Breadcrumbs active="Payer Space" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render crumbs', () => {
    const crumbs = [
      {
        name: 'my grand parent page',
        url: '/parent-page',
      },
      {
        name: 'my parent page',
        url: '/grand-parent-page',
      },
    ];
    const component = renderer.create(
      <Breadcrumbs crumbs={crumbs} active="Payer Space" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render crumbs with missing values', () => {
    const crumbs = [
      {
        name: 'my grand parent page',
        url: '/parent-page',
      },
      {
        name: '',
        url: '',
      },
    ];
    const component = renderer.create(
      <Breadcrumbs crumbs={crumbs} active="Payer Space" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render custom ellipse', () => {
    const crumbs = [
      {
        name: '',
        url: '',
      },
    ];
    const component = renderer.create(
      <Breadcrumbs crumbs={crumbs} emptyState="???" active="Payer Space" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
