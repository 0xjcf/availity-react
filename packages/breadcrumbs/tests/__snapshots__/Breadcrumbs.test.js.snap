// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render 1`] = `
<nav
  aria-label="breadcrumb"
  className=""
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/public/apps/dashboard"
      >
        Home
      </a>
    </li>
    <li
      aria-current="page"
      className="active breadcrumb-item"
    >
      …
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs should render child as current page 1`] = `
<nav
  aria-label="breadcrumb"
  className=""
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/public/apps/dashboard"
      >
        Home
      </a>
    </li>
    <li
      aria-current="page"
      className="active breadcrumb-item"
    >
      Payer Space
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs should render crumbs 1`] = `
<nav
  aria-label="breadcrumb"
  className=""
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/public/apps/dashboard"
      >
        Home
      </a>
    </li>
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/parent-page"
      >
        my grand parent page
      </a>
    </li>
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/grand-parent-page"
      >
        my parent page
      </a>
    </li>
    <li
      aria-current="page"
      className="active breadcrumb-item"
    >
      Payer Space
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs should render crumbs with missing values 1`] = `
<nav
  aria-label="breadcrumb"
  className=""
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/public/apps/dashboard"
      >
        Home
      </a>
    </li>
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/parent-page"
      >
        my grand parent page
      </a>
    </li>
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <span>
        …
      </span>
    </li>
    <li
      aria-current="page"
      className="active breadcrumb-item"
    >
      Payer Space
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs should render custom ellipse 1`] = `
<nav
  aria-label="breadcrumb"
  className=""
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <a
        href="/public/apps/dashboard"
      >
        Home
      </a>
    </li>
    <li
      aria-current={undefined}
      className="breadcrumb-item"
    >
      <span>
        ???
      </span>
    </li>
    <li
      aria-current="page"
      className="active breadcrumb-item"
    >
      Payer Space
    </li>
  </ol>
</nav>
`;
